### DISCUSSION TITLE
Changing the user and email of a past commit

### DISCUSSION THREAD

**Discussion by david-engelmann** [community/community#38097] [Other Feature Feedback, Questions, & Ideas] [answered]
While working on configuring vim, I tested a commit from vim-fugitive. For some reason it used root as the committer info (See image below).
![image](https://user-images.githubusercontent.com/81196676/199565184-0df262a4-fcad-4eab-8db2-2bbf2d6e8f30.png)
 
[More details can be found in the .patch](https://github.com/david-engelmann/DoccanoClient.jl/commit/287e56e97cb09d1856aa7d33d516514f6d127410.patch).

If it was a regular email that I accidently committed with, I could attach the email to my account ([see solved discussion](https://github.com/community/community/discussions/37630)). However in this case, I can't just add it as an additional email to my account. Is it possible to go in and update the user and email after committing to reflect my user and email?

**Comment by pameladuff** [comment]
Sorry I can't help you out. But wondering if you know how I go about getting my wallet Bitcoin off the actual network I mined along time ago.

**Reply by david-engelmann** [reply]
If you don't have your secret keys, it's called a Satoshi Donation

**Reply by pameladuff** [reply]
I have secret keys it's just been a long time. Last time I looked it was at 0 lol 😂. And Satoshi donation?? I am the original one mine on here just so much has changed can't believe all the apps now.

**Comment by airtower-luna** [comment, marked as answer]
If you have the ability to force-push to the relevant branch(es), yes. Relevant warnings first: :warning: 

1. Force-pushing will overwrite existing history, double check if what you're about to push is correct or you might lose data.
2. Changing history that others already have (outside feature branches) will likely lead to trouble, because everyone would have to manually switch to the new history.

You can use `git commit --amend` to change the current HEAD commit, and change author/committer when you do. If you don't care about keeping the timestamp intact, adding [--reset-author](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---reset-author) is the easiest option.

If you want to change only the latest commit on a branch that's all you need. Otherwise you'll need to *rebase* the branch, and mark the commit(s) you want to edit for, well, editing, and *then* amend when you get to them during the rebase. :slightly_smiling_face: See [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).

**Reply by david-engelmann** [reply]
Because there isn't any harm in having a commit that's not attributed to anyone, I'll probably just leave it. However, this was exactly what I was looking for!

**Reply by pameladuff** [reply]
I know my mnemonic and most of my optional passphrase on iancoleman I just need a reminder how to use BTC recovery so I can bruteforce my phasephrase please and thanks

**Comment by pameladuff** [comment]
I have lots of Bitcoin will give you 1 if you can help me get in.

Sent from the all new AOL app for Android 
 
  On Wed, Nov 2, 2022 at 2:06 PM, David ***@***.***> wrote:   


If you don't have your secret keys, it's called a Satoshi Donation
